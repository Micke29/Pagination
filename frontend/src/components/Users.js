import PropTypes from 'prop-types'
import React from 'react'

const Users = ({ users, loading }) => (
    loading === true ? <h2>Loading...</h2> :
        (<ul className="list-group mb-4">
            {users.map(({ id, name }) => (
                <li key={id} className="list-group-item">
                    {name}
                </li>
            ))}
        </ul>)
)

Users.propTypes = {
    users: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number.isRequired,
            name: PropTypes.string.isRequired,
        })
    ).isRequired,
    loading: PropTypes.oneOf([
        true,
        false
    ]).isRequired,
}

export default Users