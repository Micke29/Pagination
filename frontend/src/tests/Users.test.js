import { expect } from 'chai'
import React from 'react'
import { shallow } from 'enzyme'

import Users from '../components/Users'

const fakeUsers = [
    {
        id: 1,
        name: "test 1"
    },
    {
        id: 2,
        name: "test 2"
    },
    {
        id: 3,
        name: "test 3"
    },
    {
        id: 4,
        name: "test 4"
    },
    {
        id: 5,
        name: "test 5"
    }
]

describe('<Users />', () => {
    it('load users', () => {
        const wrapper = shallow(<Users users={[]} loading={true} />)

        expect(wrapper.text()).to.match(/Loading.../)
    })

    it('has 5 users', () => {
        const wrapper = shallow(<Users users={fakeUsers} loading={false} />)

        expect(wrapper.find('li')).to.have.length(5)
    })

    it('should be match its reference snapshot', () => {
        const wrapper = shallow(<Users users={fakeUsers} loading={false} />)

        expect(wrapper).to.matchSnapshot()
    })
})