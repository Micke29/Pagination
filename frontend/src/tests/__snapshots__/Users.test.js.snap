// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> should be match its reference snapshot 1`] = `
<ul
  className="list-group mb-4"
>
  <li
    className="list-group-item"
    key="1"
  >
    test 1
  </li>
  <li
    className="list-group-item"
    key="2"
  >
    test 2
  </li>
  <li
    className="list-group-item"
    key="3"
  >
    test 3
  </li>
  <li
    className="list-group-item"
    key="4"
  >
    test 4
  </li>
  <li
    className="list-group-item"
    key="5"
  >
    test 5
  </li>
</ul>
`;
